Source/JavaScriptCore/jit/JITOpcodes.cpp

```diff
@@ -777,6 +777,40 @@ void JIT::compileOpStrictEq(const JSInstruction* currentInstruction)
     VirtualRegister src1 = bytecode.m_lhs;
     VirtualRegister src2 = bytecode.m_rhs;
 
+    auto tryGetBitwiseComparableConstant = [&](VirtualRegister src) -> JSValue {
+        if (!src.isConstant())
+            return { };
+        if (!m_profiledCodeBlock->isConstantOwnedByUnlinkedCodeBlock(src))
+            return { };
+        JSValue value = m_unlinkedCodeBlock->getConstant(src);
+        if (value.isUndefinedOrNull())
+            return value;
+        if (value.isBoolean())
+            return value;
+        return { };
+    };
+
+    auto emitBitwiseComparableConstantFastPath = [&](GPRReg operandGPR, JSValue value) {
+        if constexpr (std::is_same_v<Op, OpStricteq>)
+            compare64(Equal, operandGPR, TrustedImm64(JSValue::encode(value)), jsRegT32.payloadGPR());
+        else
```

Comment:
Can we add `static_assert(<std::is_same_v<Op, OpNstricteq>);`? Also, nit but why is it `OpNstricteq`/`OpJstricteq` and not `OpNstrictEq`/`OpJstrictEq`?

Reply:
This is because the opcode is op_jstricteq, not op_jstrict_eq.